### DISCUSSION TITLE
IngressClass Configuration

### DISCUSSION THREAD

**Discussion by abdulvirta** [kubeflow/kfserving#2221] [Q&A]
Is there any document on IngressClass configuration in KServe. I'm not finding any documentation. I'm hoping to use that to prevent kserve from creating ingress controllers

**Comment by yuzisun** [comment]
We do not have a config to control this currently, but I think this is reasonable ask. I assume you are talking about the Raw deployment mode?

**Reply by abdulvirta** [reply]
yes.

Is there anyway in KServe to just tell it not to create any Ingress resources at all?

**Comment by abdulvirta** [comment]
Also I have been trying to configure the ingressClassName through this:

```
  ingress: |-
    {
        "ingressGateway" : "virta-system/private-gateway",
        "ingressService" : "virta-ingressgateway-private.virta-system.svc.cluster.local",
        "localGateway" : "knative-serving/knative-local-gateway",
        "localGatewayService" : "knative-local-gateway.istio-system.svc.cluster.local",
        "ingressDomain"  : "virta.dev",
        "ingressClassName" : "istio", # maybe point to another controller
        "domainTemplate": "{{ .Name }}-{{ .Namespace }}.{{ .IngressDomain }}"
    }
    
```

   But as far as I see, KServe does not honor the ingressClassName at all. It doesn't even add ingressClassName to the ingress that it creates. Is there a bug there? Or am missing something?
